assembly/__tests__/u256.spec.ts

```diff
@@ -348,4 +349,74 @@ describe("Basic Operations", () => {
     var r = new u256(3, 4, 6, 8);
     expect(a - b).toStrictEqual(r);
   });
+
+  it("Should multiply two u256 numbers", () => {
+    var a = u256.from(3);
+    var b = u256.from(3);
+    var r = u256.from(9);
+
+    expect(a * b).toStrictEqual(r);
+    expect(b * a).toStrictEqual(r);
+  });
+
+  it("Should multiply two u256 numbers", () => {
+    var a = u256.from(43545453452);
+    var b = u256.from(2353454354);
+    var r = new u256(10248516654965971928, 5);
+
+    expect(a * b).toStrictEqual(r);
+    expect(b * a).toStrictEqual(r);
+  });
+
+  it("Should multiply two random u256 numbers", () => {
+    let i = 0;
+    while(i< 20) {
+      const randa = <u64>Math.trunc(Math.random() * 0xFFFF)
+      const a = u256.from(randa);
+      const randb = <u64>Math.trunc(Math.random() * 0xFFFF)
+      const b = u256.from(randb);
+      const r = new u256(randa * randb);
+
+      expect(a * b).toStrictEqual(r);
+      expect(b * a).toStrictEqual(r);
+      i++;
+    }
+  });
```

Comment:
This randomly tests only lowest `u64` part with range `0 - 0xFFFF`. Which doesn't make sense to me

Reply:
It would be nice to have some manual (not randomized) tests for u256 that cover carry bits in all nibs (u64 parts). And also tests with overflow of final result